engine/verification/Readme.md

```diff
@@ -0,0 +1,170 @@
+# Verification Node
+The Verification Node in the Flow blockchain network is a critical component responsible for
+verifying `ExecutionResult`s and generating `ResultApproval`s. 
+Its primary role is to ensure the integrity and validity of block execution by performing verification processes.
+In a nutshell, the Verification Node is responsible for the following:
+1. Following the chain for new finalized blocks (`Follower` engine).
+2. Processing the execution results in the finalized blocks and determining assigned chunks to the node (`Assigner` engine).
+3. Requesting chunk data pack from Execution Nodes for the assigned chunks (`Fetcher` and `Requester` engines).
+4. Verifying the assigned chunks and emitting `ResultApproval`s for the verified chunks to Consensus Nodes (`Verifier` engine).
+![architecture.png](architecture.png)
+
+   
+## Block Consumer ([consumer.go](verification%2Fassigner%2Fblockconsumer%2Fconsumer.go))
+The `blockconsumer` package efficiently manages the processing of finalized blocks in Verification Node of Flow blockchain.
+Specifically, it listens for notifications from the `Follower` engine regarding finalized blocks, and systematically 
+queues these blocks for processing. The package employs parallel workers, each an instance of the `Assigner` engine, 
+to fetch and process blocks from the queue. The `BlockConsumer` diligently coordinates this process by only assigning 
+a new block to a worker once it has completed processing its current block and signaled its availability. 
+This ensures that the processing is not only methodical but also resilient to any node crashes. 
+In case of a crash, the `BlockConsumer` resumes from where it left off, reassigning blocks from the queue to workers,
```

Comment:
```suggestion
In case of a crash, the `BlockConsumer` resumes from where it left off by reading the processed block index from storage, reassigning blocks from the queue to workers,
```